streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -841,6 +841,8 @@ public class StreamsConfig extends AbstractConfig {
     public static final String WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG = "windowstore.changelog.additional.retention.ms";
     private static final String WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_DOC = "Added to a windows maintainMs to ensure data is not deleted from the log prematurely. Allows for clock drift. Default is 1 day";
 
+    private static final String[] NOT_CONFIGURABLE_IN_GLOBAL_CONSUMER =
+        new String[] {ConsumerConfig.GROUP_PROTOCOL_CONFIG};
```

Comment:
Why is this needed for the global consumer but not for the restore consumer? Both are manually assigned.

Reply:
Ah, I think I misunderstood. You did not mean the global consumer here, did you?

Reply:
If I indeed misunderstood, you should choose a different name.

Reply:
yeah it was called "global" in another place, I was also wondering about the name. I agree global is confusing.

Howabout "ROOT"? the root of the property tree

Reply:
What about `UNSETTABLE_COMMON_CONSUMER_CONFIGS`?

Reply:
Or `NON_SETABLE_COMMON_CONSUMER_CONFIGS`?

Reply:
IMO, `ROOT_SCOPE` is really hard to understand. It does also not follow the `NOT_CONFIGURABLE` (actually it should be `NON_CONFIGURABLE`).

But I like `IGNORED`. What about `IGNORED_COMMON_CONSUMER_CONFIGS`?

Reply:
It changes the meaning. as far as I understand this class, common consumer config are defined in `consumer.group.protocol`, but what we are ignoring is just `group.protocol`. So I think this would be incorrect

Reply:
I understood now, what you mean. This is quite confusing, because the Streams config shadows the consumer config. Although, the configs have the same name, they are used in different places. We do not want to pass `group.protocol` to the consumer. Other unprefixed consumer configs we want to pass into the consumer. 🤯 
What about `IGNORED_UNPREFIXED_CONSUMER_CONFIGS`?
That is my last suggestion. Take it or use `CONSUMER_CONFIGS_IGNORED_IN_ROOT_SCOPE`.

Reply:
sure, lets use unprefixed